### PR TITLE
[ControlledBuffer] fix logic, closes #1642

### LINKED ISSUE #1642: Controlled Buffer component’s output is "E" when input is U and buffer is enabled
I consider this a bug.

If a Controlled Buffer component's input pin is floating (U), it generates an error (E) on its output pin if the buffer is enabled.

The correct behavior would be to generate U on the buffer's output pin when its input pin is U and the buffer is enabled.  In other words, the buffer should always pass the state of its input pin (0, 1 or U) to its output pin when enabled.

The only time the buffer should generate E on its output pin is when its Enable pin is disconnected or its input pin is E.

This bug makes it impossible to create a bi-directional buffer using two controlled buffers with their outputs tied to each other's inputs (like the 74245 8-bit bus transceiver).

Interestingly, the latest version of L-E (8.03) includes a custom 74245 IC that works correctly (both the A and B sides can both be floating, with no errors when enabled). But I can't build that same IC using L-E's Controlled Buffer components because of this bug.

### PR DESCRIPTION
**Pull request by seanpquinn**
- The TRUE inversion case had a bug(?) that failed for ERROR and UNKNOWN types
- Changed ERROR branch to only handle error, not ERROR || UNKNOWN
- Updated documentation truth table to agree with the behavior described in the text

# Test plan
I looked at all input permutations, and observed behavior now matches the documentation: [test_controlled_buff_fix.zip](https://github.com/logisim-evolution/logisim-evolution/files/11133310/test_controlled_buff_fix.zip)

image

image

### CODE REVIEW

**Comment by R3dst0ne** [comment]
Please add this change to [CHANGES.md](https://github.com/logisim-evolution/logisim-evolution/blob/develop/CHANGES.md).

**Review by maehne — CHANGES_REQUESTED** [review]
@seanpquinn: Thanks for addressing this issue. Only some additional details should be fixed befor this can be merged. I agree with @R3dst0ne that this fix should be added to `CHANGES.md`.

@BFH-ktt1: It would be good if you could also have a look.

File: src/main/java/com/cburch/logisim/std/gates/ControlledBuffer.java [hunk 1]
```diff
@@ -247,11 +247,17 @@ public void propagate(InstanceState state) {
...
-      /* in cannot passed directly, as the simulator would pass an u at the input to an u at the output.
-       * Doing double inversion is the correct way to resolve this problem.
-       */
-      state.setPort(0, isInverter ? in.not() : in.not().not(), GateAttributes.DELAY);
-    } else if (control == Value.ERROR || control == Value.UNKNOWN) {
+      if (in == Value.TRUE || in == Value.FALSE) {
+        /* in cannot be passed directly, as the simulator would pass an U at the input to an U at the output.
+         * Doing double inversion is the correct way to resolve this problem.
+         */
+        state.setPort(0, isInverter ? in.not() : in.not().not(), GateAttributes.DELAY);
```
[the remark this answers is not in the document]
Comment (seanpquinn) [reply src/main/java/com/cburch/logisim/std/gates/ControlledBuffer.java:254]: Sounds good. I removed the comment originally: found it a little confusing, but put it back as I worried about breaking the simulator.

**Review by maehne — APPROVED** [review]
@seanpquinn: Thanks for addressing my feedback! Looks good to me now. However, @BFH-ktt1 should have a second look and decide whether this behavioural change is OK with him, too.

**Pull request merged**